### PR TITLE
Updated logic to handle unread messages case

### PR DESCRIPTION
**Pull request by gedu**
### Fixed Issues

$ https://github.com/Expensify/App/issues/23171
PROPOSAL: -

### Tests

- [ ] Verify that no errors appear in the JS console

1. have 2 accounts
2. with user A, mark a message as unread
3. with user B, start sending many messages
4. user A should see the unread marker in the same message
5. user A can send a message and the marker still showing

### QA Steps

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/1676818/c89d3d5c-2b25-4f57-b4d8-11e606c28ab0

https://github.com/Expensify/App/assets/1676818/f72a5410-9b73-43c6-a9c4-4a86e983a04e

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/1676818/091c6a92-3abe-4aa3-9133-cd50834b72c2

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/1676818/b1a457c8-f615-475c-a0da-1d23ff4c1a1d

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/1676818/f37c6291-5b13-46e9-9274-be66e56576b9

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/1676818/0cc57546-665d-4cbd-8f14-c8ca21bd179c

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/1676818/600fa28c-51ce-4314-af2b-8efe47dc1620

</details>

### CODE REVIEW

**Review by MonilBhavsar — COMMENTED** [review]
Some issues with marker not being displayed when revisiting the chat after marking it as unread. DM'ed Eduardo

**Comment by MonilBhavsar** [comment]
@allroundexperts could you please take a look.
@gedu there are conflicts again 😬

**Comment by gedu** [comment]
@MonilBhavsar conflicts solved

File: src/pages/home/report/ReportActionsList.js
Comment (gedu) [reply src/pages/home/report/ReportActionsList.js:214]: Was added [here](https://github.com/Expensify/App/pull/26173) point 3. 
I just moved the code around because of conflicts. Basically setting a new number produces a rerender, but when is clean, setting 0, and we try to set 0 again, because it is 0 already no rerender event is fired

File: src/libs/actions/Report.js
Comment (gedu) [reply src/libs/actions/Report.js:846]: it was for emitting the even, but I will move the emitting into the function, so it won't need to return it

**Comment by gedu** [comment]
@allroundexperts Conflicts solved + comments

**Comment by MonilBhavsar** [comment]
Thanks @gedu
@allroundexperts on you now!

**Comment by MonilBhavsar** [comment]
@allroundexperts 👋 
And, we have conflicts again :/

**Comment by gedu** [comment]
@MonilBhavsar @allroundexperts conflicts solved

**Comment by gedu** [comment]
@allroundexperts fixed

**Comment by MonilBhavsar** [comment]
@gedu could you also please add tests and QA steps and screenshot

**Comment by gedu** [comment]
@MonilBhavsar description updated

**Comment by gedu** [comment]
@allroundexperts fixed a new merge conflict, not seeing those warnings and errors.

4. I think is not related to unread marker, if that is an issue I would report it as a bug and create a new issue for it

**Comment by gedu** [comment]
@allroundexperts @MonilBhavsar Finally all good 😁

**Comment by MonilBhavsar** [comment]
Thanks! @allroundexperts could you please give it a review when you get a chance. Let's try to merge it asap

**Comment by MonilBhavsar** [comment]
Oops, conflicts again.
@allroundexperts could you please take a look when you get a chance. https://github.com/Expensify/App/pull/25935#issuecomment-1772191089

**Comment by gedu** [comment]
@MonilBhavsar and @allroundexperts all conflicts solved

**Comment by gedu** [comment]
@MonilBhavsar and @allroundexperts all new conflicts solved

**Comment by gedu** [comment]
@allroundexperts 
the green line staying even after you send a message, it is ok, look at Slack, same behavior.
I can't reproduce it, if you can provide some steps

https://github.com/Expensify/App/assets/1676818/0dc04a53-35e0-4a4c-9eaa-a326ae1d3a92


FYI, can you tests on Chrome or Safari, I can not assure the behavior on Brave or other browsers, our main browser are Chrome and Safari for what I understand

**Comment by gedu** [comment]
@allroundexperts all should be good now

**Comment by MonilBhavsar** [comment]
Conflicts again and @allroundexperts 👋

**Comment by gedu** [comment]
@MonilBhavsar @allroundexperts all should be ok

**Comment by gedu** [comment]
@MonilBhavsar @mountiny Bumping

**Comment by MonilBhavsar** [comment]
I bumped internal thread https://expensify.slack.com/archives/C02NK2DQWUX/p1699350386425929?thread_ts=1699018532.246269&cid=C02NK2DQWUX

**Comment by shubham1206agra** [comment]
Will upload the checklist today

**Comment by shubham1206agra** [comment]
@gedu Can you check this when I moved away from report, the report is getting read somehow?
https://github.com/Expensify/App/assets/58412969/a5768752-13b6-4945-ae3f-a81704669beb

**Comment by shubham1206agra** [comment]
Maybe merge main (not sure)

**Comment by gedu** [comment]
> @gedu Can you check this when I moved away from report, the report is getting read somehow? cc @MonilBhavsar

@shubham1206agra does it happen also on main? I think I saw it before

**Comment by shubham1206agra** [comment]
prod doesn't have this problem

**Comment by shubham1206agra** [comment]
> > @gedu Can you check this when I moved away from report, the report is getting read somehow? cc @MonilBhavsar
> 
> @shubham1206agra does it happen also on main? I think I saw it before

Same I saw this earlier

**Comment by shubham1206agra** [comment]
Main doesn't have this problem

**Comment by gedu** [comment]
Just merged with main

**Comment by shubham1206agra** [comment]
@gedu Problem still persists. Not repro on main

**Comment by gedu** [comment]
https://github.com/Expensify/App/assets/1676818/3d54a7cd-d952-4cbc-8ca9-40e03d3f191e

@shubham1206agra I just reproduce it on `main` it is auto marked as read

**Comment by shubham1206agra** [comment]
This is really weird. But if that's the case, I am going to ignore this bug and will upload the checklist today.
@MonilBhavsar Do we have any GH issue for this bug?

**Comment by shubham1206agra** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/58412969/a3c4e925-35d4-48a3-a4ff-39167ee74a47

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/58412969/e81d0aad-9e55-4809-b95f-b242fe65ce89

</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/58412969/d1bffd2f-c27c-40ef-a145-6bbefbfda9e2

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/58412969/477f3759-e6fe-4fc0-8c2f-1f48d1159ef3

</details>

File: ios/Podfile.lock
Comment (shubham1206agra) [inline ios/Podfile.lock:1302]: You may want to remove this

**Review by shubham1206agra — APPROVED** [review]

File: ios/Podfile.lock
Comment (MonilBhavsar) [reply ios/Podfile.lock:1302]: +1

File: src/components/EmojiPicker/EmojiPicker.js
Comment (MonilBhavsar) [inline src/components/EmojiPicker/EmojiPicker.js:137]: Why this change?

File: src/pages/home/report/ReportActionsList.js
Comment (MonilBhavsar) [inline src/pages/home/report/ReportActionsList.js:329]: NAB, we can remove this

File: tests/ui/UnreadIndicatorsTest.js
Comment (MonilBhavsar) [inline tests/ui/UnreadIndicatorsTest.js:271]: > given that the test doesn't behave as the app
What does this mean?

File: src/pages/home/report/ReportActionsList.js
Comment (MonilBhavsar) [inline src/pages/home/report/ReportActionsList.js:208]: where is messageManuallyMarkedUnread initialized?

File: src/pages/home/report/ReportActionsList.js
Comment (MonilBhavsar) [inline src/pages/home/report/ReportActionsList.js:153]: Why we need to store this in ref. Can we access report.lastReadTime directly?

File: src/pages/home/report/ReportActionsList.js
Comment (MonilBhavsar) [inline src/pages/home/report/ReportActionsList.js:225]:
```suggestion
        // Listen to specific reportID for unread event and set the marker to new message
```

File: src/pages/home/report/ReportActionsList.js
Comment (MonilBhavsar) [inline src/pages/home/report/ReportActionsList.js:334]: NAB, we can remove this line

File: src/pages/home/report/ReportActionsList.js
Comment (MonilBhavsar) [inline src/pages/home/report/ReportActionsList.js:344]: NAB, we can remove this line

File: src/pages/home/report/ReportActionsList.js
Comment (shubham1206agra) [reply src/pages/home/report/ReportActionsList.js:208]: `const [messageManuallyMarkedUnread, setMessageManuallyMarkedUnread] = useState(0);` Line 159

File: src/pages/home/report/ReportActionsList.js
Comment (shubham1206agra) [reply src/pages/home/report/ReportActionsList.js:153]: See Line 208

File: src/pages/home/report/ReportActionsList.js
Comment (MonilBhavsar) [reply src/pages/home/report/ReportActionsList.js:208]: Thanks! I swear, I searched

**Comment by shubham1206agra** [comment]
@gedu Friendly bump

File: src/components/EmojiPicker/EmojiPicker.js
Comment (gedu) [reply src/components/EmojiPicker/EmojiPicker.js:137]: One of the tests was throwing a lot of errors because of the listener was null, was impossible to fix the unite tests from ReportActionsList without this
@MonilBhavsar

File: tests/ui/UnreadIndicatorsTest.js
Comment (gedu) [reply tests/ui/UnreadIndicatorsTest.js:271]: When there is a single panel, in the tests, seems that it keeps all components in memory, so I need to reset the state to make it work, "simulating" the same as what the app does

**Comment by gedu** [comment]
@shubham1206agra updated

File: src/components/EmojiPicker/EmojiPicker.js
Comment (MonilBhavsar) [reply src/components/EmojiPicker/EmojiPicker.js:137]: Thanks for clarifying 👍

File: src/pages/home/report/ReportActionsList.js
Comment (MonilBhavsar) [inline src/pages/home/report/ReportActionsList.js:95]:
```suggestion
// We need to persist it across reports because there are at least 3 ReportScreen components created so the
```

**Comment by MonilBhavsar** [comment]
> This is really weird. But if that's the case, I am going to ignore this bug and will upload the checklist today.
@MonilBhavsar Do we have any GH issue for this bug?

@shubham1206agra could you please mention the steps to reproduce the issue

**Comment by shubham1206agra** [comment]
> > This is really weird. But if that's the case, I am going to ignore this bug and will upload the checklist today.
> > @MonilBhavsar Do we have any GH issue for this bug?
> 
> @shubham1206agra could you please mention the steps to reproduce the issue

Requires 2 users

User A:
1. Select 1:1 chat with user B.
2. Go to other chat

Now, with user B, start sending messages to A.

**Comment by MonilBhavsar** [comment]
Discussed 1:1 with Eduardo, looks like updating prevReportID to useRef is causing this issue. He is going to take a look and confirm

**Comment by gedu** [comment]
I just took a look at it, and the bug wasn't coming from our code, was coming from `PusherUtils`, but after merging again with `main` I couldn't reproduce the bug. It is all pushed, if you can give it another round. @MonilBhavsar @shubham1206agra 

https://github.com/Expensify/App/assets/1676818/4a07824e-4fe8-4323-8ec9-171707825dea

**Comment by shubham1206agra** [comment]
Have you pushed the merge commit? @gedu 
Cause I can still able to repro this on this PR.

**Comment by gedu** [comment]
@shubham1206agra @MonilBhavsar pushed the merge

**Comment by gedu** [comment]
Checking again, because @MonilBhavsar can reproduce it, and I just could reproduce it again

**Comment by shubham1206agra** [comment]
@gedu Can you check the repro on `main`?

**Comment by MonilBhavsar** [comment]
It is not reproducible on main for me

**Comment by MonilBhavsar** [comment]
Thanks! It works fine for me now. @shubham1206agra could you also please confirm, thanks

**Comment by shubham1206agra** [comment]
@MonilBhavsar Working fine now

**Comment by MonilBhavsar** [comment]
Wohoo, thanks for confirming!

**Review by MonilBhavsar — APPROVED** [review]
Looks good! 🚀 
Thanks all!

**Comment by MonilBhavsar** [comment]
Dismissed old PR reviewer checklist.
New one is here https://github.com/Expensify/App/pull/25935#issuecomment-1804272308

**Comment by rlinoz** [comment]
This PR might have caused [this regression](https://github.com/Expensify/App/issues/31617), generating an adhoc build to check.

**Comment by situchan** [comment]
Potential regressions:
- https://github.com/Expensify/App/issues/31617
- https://github.com/Expensify/App/issues/31621
- https://github.com/Expensify/App/issues/31623

**Comment by rlinoz** [comment]
From what I tested it potentially is, but I didn't actually revert this PR to check, just tested the Adhoc build.

File: src/pages/home/report/ReportActionsList.js
Comment (arosiclair) [inline src/pages/home/report/ReportActionsList.js:97]: @gedu @MonilBhavsar Hey I'm working on a bug related to unread markers and I'm having trouble wrapping my head around this cache. What problem/scenario does it fix specifically? Like is there a scenario where when we switch away from a report and then navigate back, we want the unread marker to persist?

**Pull request merged**
